docs/release/handbook.md

```diff
@@ -0,0 +1,288 @@
+# Kubeflow Release Handbook
+
+## Executive Summary
+
+The purpose of this document is to define the roles and processes for the Kubeflow community to release Kubeflow. This document will attempt to coordinate releasing between the five major components and working groups in Kubeflow, as well as defining best practices, roles, and processes to make releasing Kubeflow easier for the community with each iteration.
+
+## People and Roles
+
+### Release Manager
+The Release Manager will be responsible for coordinating the release and taking ultimate accountability for all release tasks to be completed on time.
+
+**Prerequisites:**
+1. Be a contributor for at least 2 years
+2. Have shadowed in a previous release or be a Release Team member in the past
+3. Contributions in at least 2 Working Groups
+
+**Time Commitments:** This role will need to be synchronous. The Release Manager is expected to be joining all the weekly and burndown meetings as well as coordinate with the Release Team and the Working Group leads. For the Development phase an estimate of the work needed is 4-8 hours per week. For the next phases the Release Manager will need to devote more time and coordinate the tasks of the release.
+
+**Responsibilities:**
+* Providing updates to the mailing list with the progress of the release
```

Comment:
Rather than only providing updates to the mailing list, it should be the responsibility to maintain a KanBan with all the PRs and issues relevant for the release from all the working groups (as was done in the past). This is a much more efficient method for tracking the release progress than a single tracking issue as you can also set milestones.  For example, this would also allow the working groups to add issues or PRs for bug fixes to the `rc1` milestone so we can easily keep track of what still needs to be done for that release.

Reply:
Updates to the mailing list should be the bare minimum.

It's then up to the release team to decide how to best track the progress of the release and expose it to the users.
Also, the proposals to have KanBan and milestones to track issues seems interesting